scanner/cmd/scanner/main.go

```diff
@@ -75,6 +82,15 @@ func main() {
 		utils.CrashOnError(os.Setenv(mtls.KeyFileEnvName, filepath.Join(p, mtls.ServiceKeyFileName)))
 	}
 
+	//  If proxy path is set, periodically check for updates.
+	if cfg.Proxy.ConfigDir != "" {
+		zlog.Info(ctx).
+			Str("dir", cfg.Proxy.ConfigDir).
+			Str("file", cfg.Proxy.ConfigFile).
+			Msg("proxy configured")
+		proxy.WatchProxyConfig(ctx, cfg.Proxy.ConfigDir, cfg.Proxy.ConfigFile, true)
```

Comment:
I don't think a restart is needed, as this should do the trick. This is the standard method other StackRox components do, so I think it's ok as-is

Reply:
The only examples I have I guess are Central and Scanner v2, since those are the only components in the CentralServices. They both use this same paradigm.

Note: for the SecuredCluster, a restart is required, as we do not use the proxy-config secret. Rather, it is up to users to update the environment variables